apps/services/user-profile/src/app/user-profile/userProfile.controller.ts

```diff
@@ -428,6 +446,8 @@ export class UserProfileController {
     if (nationalId != user.nationalId) {
       throw new BadRequestException()
     } else {
+      // findOrCreateUserProfile for edge cases - fragmented onboarding
+      await this.findOrCreateUserProfile(nationalId, user)
```

Comment:
Why do this on device token delete? Shouldn't the profile exists if device token exists?

Reply:
true - removing as we speak